### PR TITLE
Implement `tables.get` rpc endpoint

### LINKED ISSUE #3611: Implement `tables.get` RPC method

### PR DESCRIPTION
**Pull request by Anish9901**
Fixes #3611

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `develop` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/rpc/tables.py [hunk 1]
```diff
@@ -47,6 +47,26 @@ def list_(*, schema_oid: int, database_id: int, **kwargs) -> list[TableInfo]:
     ]
 
 
+@rpc_method(name="tables.get")
+@http_basic_auth_login_required
+@handle_rpc_exceptions
+def get_(*, table_oid: int, database_id: int, **kwargs) -> TableInfo:
```
Comment (mathemancer) [inline mathesar/rpc/tables.py:53]: I don't understand why you went with `get_` here. `get` isn't a keyword, is it?

File: mathesar/rpc/tables.py [hunk 2]
```diff
@@ -47,6 +47,26 @@ def list_(*, schema_oid: int, database_id: int, **kwargs) -> list[TableInfo]:
...
+    """
+    List information about a table for a schema. Exposed as `get_`.
```
Comment (mathemancer) [inline mathesar/rpc/tables.py:55]: Should be exposed as `get` regardless.

**Review by mathemancer — CHANGES_REQUESTED** [review]
I have either one change to request, or a choice you made that I need to have explained to me: Why use `get_` rather than `get` in the RPC function?
- It makes the docs a little wonky, and
- Using the trailing underscore is typically reserved for situations where you want to avoid colliding with a keyword in python.

File: mathesar/rpc/tables.py (hunk 1 above)
Comment (Anish9901) [reply mathesar/rpc/tables.py]: You are right, I erroneously thought it was a keyword. I've changed it to `get`.

**Review by mathemancer — APPROVED** [review]
Okay, I think this is good to go once the tests pass.

**Pull request merged**